opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -61,15 +62,13 @@ namespace Opm
             allcells_[cell] = cell;
         }
 #if HAVE_MPI
-        if ( terminal_output_ ) {
-            if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
-            {
-                const ParallelISTLInformation& info =
-                    boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
-                // Only rank 0 does print to std::cout
-                terminal_output_ = ( info.communicator().rank() == 0 );
-                is_parallel_run_ = ( info.communicator().size() > 1 );
-            }
+        if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
+        {
+            const ParallelISTLInformation& info =
+                boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
+            // Only rank 0 does print to std::cout
+            terminal_output_ &= ( info.communicator().rank() == 0 );
```

Comment:
> I tested it and it worked

On your version of your compiler today.  Using the bitwise operator assumes a lot about the underlying representation of `bool`s so you really need an operator that applies to objects rather than bits.

> Do you mean it might fail with different compilers?

Yes, that is what I mean.